pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -333,24 +335,21 @@ func (nim *nodeInfoManager) updateCSINodeInfo(
 	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		nodeInfo, err := csiKubeClient.CsiV1alpha1().CSINodeInfos().Get(string(nim.nodeName), metav1.GetOptions{})
 		if nodeInfo == nil || errors.IsNotFound(err) {
-			return nim.createNodeInfoObject(driverName, driverNodeID, topology)
+			err = nim.CreateCSINodeInfo()
```

Comment:
If this succeeds, you are going to pass a nil nodeInfo to installDriverToCSINodeInfo. It looks like that will cause a nil deref when you loop over nodeInfo.CSIDrivers